packages/web/app/components/ProcedureModal.jsx

```diff
@@ -25,77 +73,192 @@ const getEndDateTime = ({ date, startTime, endTime }) => {
   return toDateTimeString(addDays(parseISO(actualEndDateTime), 1));
 };
 
+const useProgramResponsesQuery = (patientId, procedureId) => {
+  const api = useApi();
+  return useQuery([], () => api.get(`patient/${patientId}/programResponses`, { procedureId }));
+};
+
 export const ProcedureModal = ({ onClose, onSaved, encounterId, editedProcedure }) => {
   const api = useApi();
+  const { currentUser } = useAuth();
+  const { patientId } = useParams();
+  const { data: patient } = usePatientDataQuery(patientId);
+  const [refreshCount, updateRefreshCount] = useRefreshCount();
   const [selectedSurveyId, setSelectedSurveyId] = useState(null);
-  const locationSuggester = useSuggester('location', {
-    baseQueryParameters: { filterByFacility: true },
-  });
-  const physicianSuggester = useSuggester('practitioner');
-  const departmentSuggester = useSuggester('department');
-  const anaesthetistSuggester = useSuggester('practitioner');
-  const assistantSuggester = useSuggester('practitioner');
-  const procedureSuggester = useSuggester('procedureType');
-  const anaestheticSuggester = useSuggester('drug');
+  const [unSavedChangesModalOpen, setUnSavedChangesModalOpen] = useState(false);
+  const [unSavedProgramFormOpen, setUnSavedProgramFormOpen] = useState(false);
+  const procedureId = editedProcedure?.id;
+  const { data: programResponses } = useProgramResponsesQuery(patientId, procedureId);
 
   return (
-    <>
-      <FormModal
-        width="md"
-        title={
-          <TranslatedText
-            stringId="procedure.modal.title"
-            fallback=":action procedure"
-            replacements={{
-              action: editedProcedure?.id ? (
+    <Form
```

Comment:
i had to move up the Form component to be outside of the modal so that I could access the 'dirty' prop when closing the modal.